components/sup/src/event.rs

```diff
@@ -177,6 +179,26 @@ pub fn service_stopped(service: &Service) {
     }
 }
 
+/// Send an event at the start of a Service update.
+pub fn service_update_started(service: &Service, update: &PackageIdent) {
+    if stream_initialized() {
+        publish(ServiceUpdateStartedEvent { event_metadata:   None,
+                                            service_metadata: Some(service.to_service_metadata()),
```

Comment:
Not new code, but it seems like instead of having `Service::to_service_metadata`, it would be more idiomatic to have `impl Into<ServiceMetadata> for &Service`, then this becomes:
```suggestion
                                            service_metadata: Some(service.into()),
```

Reply:
@baumanj yep, makes total sense. i'll refactor in a separate pr.